src/lib/translators/mongo.ts

```diff
@@ -222,6 +223,43 @@ function transformTop(step: TopStep): Array<MongoStep> {
   ];
 }
 
+/** transform an 'unpivot' step into corresponding mongo steps */
+function transformUnpivot(step: UnpivotStep): Array<MongoStep> {
+  let mongoPipeline: Array<MongoStep> = [];
+  let projectCols: PropMap<string> = {};
+  let objectToArray: PropMap<string> = {};
+
+  // Prepare the projectCols to be included in Mongo $project steps
+  for (const col of step.keep) {
+    projectCols[col] = `$${col}`;
+  }
+
+  // Prepare the objectToArray object to be included in the first Mongo $project step
+  for (const col of step.unpivot) {
```

Comment:
Same comment as above:

```typescript
import _ from 'lodash';

const objectToArray: PropMap<string> = _.fromPairs(step.unpicot.map(col => [col, `$${col}`]));
```

I even wonder if this specific oneliner should not be in a utils function since we do this kind of mapping (`col` -> `$col`) quite often.

Reply:
I have a small ongoing PR about that.